ObjectiveCloudantTests/QueryViewTests.m

```diff
@@ -0,0 +1,971 @@
+//
+//  QueryViewTests.m
+//  ObjectiveCloudant
+//
+//  Created by Rhys Short on 15/02/2016.
+//  Copyright (c) 2016 IBM Corp.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+//  except in compliance with the License. You may obtain a copy of the License at
+//    http://www.apache.org/licenses/LICENSE-2.0
+//  Unless required by applicable law or agreed to in writing, software distributed under the
+//  License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+//  either express or implied. See the License for the specific language governing permissions
+//  and limitations under the License.
+//
+
+#import <XCTest/XCTest.h>
+#import <ObjectiveCloudant/ObjectiveCloudant.h>
+#import "TestHelpers.h"
+
+@interface QueryViewTests : XCTestCase
+@property (nonatomic, strong) NSString *url;
+@property (nonatomic, strong) NSString *dbName;
+@property (nonatomic, strong) NSString *username;
+@property (nonatomic, strong) NSString *password;
+@property (nonatomic, strong) CDTDatabase *database;
+@end
+
+@implementation QueryViewTests
+
+- (void)setUp
+{
+    [super setUp];
+
+    self.url = @"http://localhost:5984";
+    self.username = nil;
+    self.password = nil;
+
+    // These tests require their own database as they modify content; create one
+
+    self.dbName = @"objectivecouch-test";
+
+    CDTCouchDBClient *client = [CDTCouchDBClient clientForURL:[NSURL URLWithString:self.url]
+                                                     username:self.username
+                                                     password:self.password];
+    self.database = client[self.dbName];
+}
+
+- (void)tearDown
+{
+    // Put teardown code here. This method is called after the invocation of each test method in the
+    // class.
+    [super tearDown];
+}
+
+- (void)testfailsWithKeyandKeysSet
```

Comment:
I think the name `testfails` is confusing. The names of tests should probably assert a positive statement. Here, it'd be something like `testUsingKeyAndKeysGivesError`. Similarly, the `testSuccessfully` prefixes are a bit redundant, as all tests should be successful in some manner. I think the `Successfully` can be removed from all of them.